pi/controller.c

```diff
@@ -79,17 +75,29 @@ static int pi_dma_read(struct pi_controller *pi) {
   else if ((source & 0x05000000) == 0x05000000)
     dd_dma_read(pi->bus->dd, source, dest, length);
 
+  // FIXME: verify this
+  pi->regs[PI_RD_LEN_REG] = 0x7F;
```

Comment:
Does this FIXME apply to all three assignments or just the PI_RD_LEN_REG one?

Reply:
all three, changed comment to `these`